constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java

```diff
@@ -2251,4 +2275,125 @@ public String taskAsJson(TaskRequest req) throws ConstructorException {
+
+            if (req.getSection() != null) {
+                url = url.newBuilder()
+                        .addQueryParameter("section", req.getSection())
+                        .build();
+            }
+
+            if (req.getVersionId() != null) {
+                url = url.newBuilder()
+                        .addQueryParameter("version_id", req.getVersionId())
+                        .build();
+            }
+
+            if (req.getA().size() > 0) {
+                for (List<String> questionAnswers : req.getA())
+                {
+                    String answerParam = String.join(",", questionAnswers);
+                    url = url.newBuilder().addQueryParameter("a", answerParam).build();
+                }
+            } else {
+                if (type.equals("finalize")) {
+                    throw new IllegalArgumentException("a (answers) is a required parameter for a finalize request");
+                }
+            }
+
+            Request request = this.makeUserRequestBuilder(userInfo)
+                    .url(url)
+                    .get()
+                    .build();
+
+            return request;
+        } catch (Exception exception) {
+            throw new ConstructorException(exception);
+        }
+    }
+
+    /**
+     * Queries the quiz service for the next quiz/question
+     *
+     * @param req the Next Quiz request
+     * @return a Next Quiz Response
+     * @throws ConstructorException if the request is invalid.
+     */
+    public NextQuizResponse nextQuiz(QuizRequest req, UserInfo userInfo) throws ConstructorException {
+        try {
+            Request request = createQuizRequest(req, "next", userInfo);
+            Response response = clientWithRetry.newCall(request).execute();
+            String json = getResponseBody(response);
+            return createNextQuizResponse(json);
+        } catch (Exception exception) {
+            throw new ConstructorException(exception);
+        }
+    }
+
+    /**
+     * Queries the quiz service for the next quiz/question
+     *
+     * @param req the Next Quiz request
+     * @return a string of JSON
+     * @throws ConstructorException if the request is invalid.
+     */
+    public String nextQuizAsJson(QuizRequest req, UserInfo userInfo) throws ConstructorException {
+        try {
+            Request request = createQuizRequest(req, "next", userInfo);
+            Response response = clientWithRetry.newCall(request).execute();
+            return getResponseBody(response);
+        } catch (Exception exception) {
+            throw new ConstructorException(exception);
+        }
+    }
+
+    /**
+     * Queries the quiz service for the next quiz/question
+     *
+     * @param req the Finalize Quiz request
+     * @return a Finalize Quiz Response
+     * @throws ConstructorException if the request is invalid.
+     */
+    public FinalizeQuizResponse finalizeQuiz(QuizRequest req, UserInfo userInfo) throws ConstructorException {
```

Comment:
Should we change `finalizeQuiz` to one of the following? 🤔
* `quizResults`
* `getQuizResults`